content/vendor/guides/configuring-application-snapshots.md

```diff
@@ -0,0 +1,345 @@
+
+{{< notes title="Note" >}}
+This yaml is trucated for readability purposes. 
+The entire file is located [here](https://github.com/replicatedhq/kotsapps/blob/master/app-direct/manifests/postgres-deployment.yaml) in the GitHub repo
+{{< /notes >}}
+
+```shell
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: app-direct-postgresql
+  labels:
+    app: postgresql
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgresql
+  serviceName: postgres-service
+  template:
+    metadata:
+      labels:
+        app: postgresql   
+    spec:
+      containers:
+...
+      volumeMounts:
+        - name: postgresql-vct
+          mountPath: /var/lib/postgresql/data/pgdata
+          subPath: postgresql-db
+  volumeClaimTemplates:
+  - metadata:
+      name: postgresql-vct
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 2Gi
+
+```
+
+To have the volume included in the snapshot, we add the label annotation as shown below:
+
+```shell
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: app-direct-postgresql
+  labels:
+    app: postgresql
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgresql
+  serviceName: postgres-service
+  template:
+    metadata:
+      labels:
+        app: postgresql 
+      annotations:
+        backup.velero.io/backup-volumes: postgresql-vct # this volume will be included in snapshot
+    spec:
+      containers:
+...
+
+```
+
+The documentation also provides an example [here](https://kots.io/vendor/snapshots/configuring-backup/#identify-volumes) 
+
+### Optional/Advanced Use Case: Backup Hooks
+
+In the example above, we are simply taking a snapshot of the volume. 
+While that may suffice for certain cases, in other cases you may need perform actions before the snapshot takes place. 
+For this, we have Backup Hooks. 
+Velero provides both pre and post [backup hooks](https://velero.io/docs/v1.4/hooks/#docs). 
+These can be configured by adding annotations to the pod itself or in the [Backup spec](https://velero.io/docs/v1.2.0/api-types/backup/). 
+For the purposes of this guide, we'll do the former and use label annotations.
+
+In the example above, a more realistic scenario is that we'll probably want to run ```pg_dump``` and only take a snapshot of that backup. 
+To accomplish this we'll need to:
+
+- Add the ```pre.hook.backup.velero.io/command``` label annotation to execute ```pg_dump``` and put the backup in a new folder.
+- Add the ```pre.hook.backup.velero.io/timeout``` label annotation to give the backup time to run and store the output.
+  Per the Velero documentation, "The hook is considered in error if the command exceeds the timeout." so make sure this value exceeds how long you expect the command to take.
+- Add a new volume to the pod and mount it to the new folder containing the ```pg_dump``` output.
+
+Below is the same definition file with the modifications:
+
+```shell
+
+apiVersion: apps/v1
```

Comment:
This file is super long and it's not super easy to eyeball the changes that have been made. While I can certainly just copy-paste what's here and probably get the guide working (I'm still running through it, I'll let you know) it would be cool to see some more details on exactly what's changing -- maybe something like what we've done with the `diff` code block type here: https://kots.io/vendor/guides/helm-chart/#map-to-helm-chart